l10n_it_ddt/models/stock_picking_package_preparation.py

```diff
@@ -435,6 +435,44 @@ def action_invoice_create(self):
                 subtype_id=self.env.ref('mail.mt_note').id)
         return [inv.id for inv in invoices.values()]
 
+    @api.multi
+    def action_send_ddt_mail(self):
+        self.ensure_one()
+        ir_model_data = self.env['ir.model.data']
+        try:
+            template_id = ir_model_data.\
+                get_object_reference('l10n_it_ddt',
+                                     'email_template_edi_ddt')[1]
+        except ValueError:
+            template_id = False
+
+        try:
+            compose_form_id = ir_model_data.\
+                get_object_reference('mail',
+                                     'email_compose_message_wizard_form')[1]
+        except ValueError:
+            compose_form_id = False
+
+        ctx = {
+            'default_model': 'stock.picking.package.preparation',
+            'default_res_id': self.ids[0],
+            'default_use_template': bool(template_id),
+            'default_template_id': template_id,
+            'mark_so_as_sent': True,
```

Comment:
This should be replaced by something like `mark_ddt_as_sent` and used properly, otherwise just remove it

Reply:
right! will have a look

Reply:
Have removed for now the mark_so_as_sent because not used/required in this particular context. I will add it as additional patch on top of this after approved ;)

Reply:
Perfect, anyway I still can't manage to see `mail_template_data_notification_email_ddt` working, any advice?

Reply:
If you have received the notification email with attached the DDT, the header and footer are exactly the template we are talking

Reply:
I'm trying this in runbot and it doesn't really send emails, is there any way to check that within Odoo (I tried in Email/Email but no luck)

Reply:
Have you tried to run it locally?

Reply:
No I haven't, but I've managed to see it rendered in runbot and it looks good.